indico/htdocs/js/indico/jquery/categorynavigator.js

```diff
@@ -643,25 +649,116 @@
 
         _onAction: function(categoryId) {
             var self = this;
-            var res = self.options.onAction(self._categories[categoryId]);
 
-            if (res === undefined) {
-                res = $.Deferred().resolve();
+            function callback() {
+                var res = self.options.onAction(self._categories[categoryId]);
+                if (res === undefined) {
+                    res = $.Deferred().resolve();
+                }
+
+                res.then(function() {
+                    if (self.dialog) {
+                        self.dialog.close();
+                    }
+                });
+            }
+
+            if (self.options.confirmation) {
+                var text = $T.gettext("You selected category {0}. Are you sure you want to proceed?").format(self._categories[categoryId].title);
```

Comment:
I'd break it, same way we do in several other places (but again, just a preference):
```js
var text = $T.gettext("You selected category {0}. Are you sure you want to proceed?")
             .format(self._categories[categoryId].title);
```